### DISCUSSION TITLE
Is it possible to determine dynamic names to execute the actions?

### DISCUSSION THREAD

**Discussion by Daespinosag** [spatie/laravel-queueable-action#63] [Q&A]
Hello, people, 

I have a problem with the standard defined in the public function of the shares.

In my team we define the `run `clause to execute a specific action. however the package only supports `__invoke `and `execute`. is there any clean way to keep using this package in this context?

- The action execution method

```php
public function run(
        ContractData $contractData
    ) : void { ... }
```

- Execution in queue

```php
$action->onQueue()->run($contractData)
```

- Error

```php
at vendor/spatie/laravel-queueable-action/src/ActionJob.php:94
     90▕ 
     91▕     public function handle()
     92▕     {
     93▕         $action = app($this->actionClass);
  ➜  94▕         $action->{$action->queueMethod()}(...$this->parameters);
     95▕     }
     96▕ 
     97▕     public function __sleep()
     98▕     {

      +20 vendor frames 

      Illuminate\Foundation\Bus\PendingDispatch::__destruct()
```
in this case it is `run `but it would be ideal to be able to leave open the possibility of using any name (`execute`, `run`, `notificate`, `generate`, ... ).

**Does anyone have an idea how to fix this ?**

**Comment by brendt** [comment]
That's not possible, and we don't have any plans on making the scope of this package larger.

**Reply by Daespinosag** [reply]
@brendt  I have the changes to support this in a branch, are you accepting collaborations? It does not change anything, it only includes the possibility of configuring the functions to be executed. [PR](https://github.com/spatie/laravel-queueable-action/pull/64)

**Reply by brendt** [reply]
Unfortunately we don't want to support that functionality, we want to keep the scope of this package small.
